src/daft-core/src/array/ops/cast.rs

```diff
@@ -442,12 +463,17 @@ fn append_values_from_numpy<
             }
         };
         values_vec.extend(sl.iter().map(|v| <Wrapping<Tgt> as NumCast>::from(*v).unwrap().0));
-        shapes_vec.push(pyarray.shape().iter().map(|v| *v as u64).collect::<Vec<u64>>());
-        Ok((sl.len()))
+        shapes_vec.extend(pyarray.shape().iter().map(|v| *v as u64));
+        Ok((sl.len(), pyarray.shape().len()))
     })
 }
 
-type ArrayPayload<Tgt> = (Vec<Tgt>, Option<Vec<i64>>, Option<Vec<Vec<u64>>>);
+type ArrayPayload<Tgt> = (
+    Vec<Tgt>,
+    Option<Vec<i64>>,
```

Comment:
should these match?

Reply:
Hmm what do you mean by "match", and to what are you referring to by "these"? The first two fields are the data array and the corresponding offsets array, where the former has a native templated type and the latter should always have an `i64` type.

The last two fields are the shape array and the corresponding offsets array.